src/control/server/mgmt_pool.go

```diff
@@ -63,19 +71,42 @@ func (svc *mgmtSvc) PoolCreate(ctx context.Context, req *mgmtpb.PoolCreateReq) (
 		return nil, FaultPoolNvmeTooSmall(req.Nvmebytes, targetCount)
 	}
 
-	dresp, err := mi.CallDrpc(drpc.ModuleMgmt, drpc.MethodPoolCreate, req)
-	if err != nil {
-		return nil, err
-	}
+	try := 0
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+		}
 
-	resp := &mgmtpb.PoolCreateResp{}
-	if err = proto.Unmarshal(dresp.Body, resp); err != nil {
-		return nil, errors.Wrap(err, "unmarshal PoolCreate response")
-	}
+		dresp, err := mi.CallDrpc(drpc.ModuleMgmt, drpc.MethodPoolCreate, req)
+		if err != nil {
+			return nil, err
+		}
 
-	svc.log.Debugf("MgmtSvc.PoolCreate dispatch, resp:%+v\n", *resp)
+		resp := &mgmtpb.PoolCreateResp{}
+		if err = proto.Unmarshal(dresp.Body, resp); err != nil {
+			return nil, errors.Wrap(err, "unmarshal PoolCreate response")
+		}
 
-	return resp, nil
+		svc.log.Debugf("MgmtSvc.PoolCreate dispatch, try %d, resp:%+v\n", try, *resp)
+
+		ds := drpc.DaosStatus(resp.GetStatus())
+		switch ds {
+		// retryable errors
+		case drpc.DaosGroupVersionMismatch:
+			svc.log.Infof("MgmtSvc.PoolCreate (try %d), retrying due to %s", try, ds)
```

Comment:
I'm wondering if we would be better served by logging the cumulative timeout instead of (or in addition to) the try count.  It might make it easier to understand the magnitude of the retry delay.

Reply:
Per @liw, this should be pretty rare and not require more than one or two retries. We'll see the timestamps in the log if this occurs, so I don't really see how adding up the elapsed time adds much.

Reply:
Why is this an info message and not a debug message?  Does the admin really need to know how many retries we did?  Are they going to start asking questions or making assumptions when they see this number fluctuate from run to run?

Reply:
It's an info-level message because it's significant enough to be logged during normal operations, but it's not necessarily an error. The intent was to leave a breadcrumb for the admin to answer the question, "Why is dmg pool create taking so long to return?"

Again, this condition should be pretty rare, as indicated by the fact that the race it's intended to address is only intermittently seen in CI. My understanding is there is only a small window of opportunity to hit it after the system map is updated (which occurs during system startup as members join the cluster, or after a rank is excluded/reintegrated).

Reply:
is there effort going into fixing the race or is it something that can't be easily fixed?

Reply:
It's not really something to be fixed. It's just that after the system map changes there's a brief window where pool creates (and possibly other pool operations) are rejected. This commit just retries on behalf of the user because we know that it's safe to retry after that particular error. I suspect that we'll find other situations like going forward.